Documentation/WorkflowSwiftUI Adoption Guide.md

```diff
@@ -0,0 +1,448 @@
+      PersonView(store: store.person2)
+    }
+  }
+}
+```
+
+Child store mapping works for plain `ObservableModel` properties, as well as optionals, collections, and identified collections.
+
+## Actions
+
+If your model is a single-action `ActionModel` created by `context.makeActionModel()` , you can send actions by calling `send` on your `Store`.
+
+For complex models with multiple sinks, you can access each sink on the model through the `Store` and send through them directly.
+
+```swift
+struct CustomModel: ObservableModel {
+  var accessor: StateAccessor<State>
+
+  var up: Sink<UpAction>
+  var down: Sink<DownAction>
+}
+
+struct CustomView: View {
+  typealias Model = CustomModel
+
+  let store: Store<Model>
+  
+  var body: some View {
+    WithPerceptionTracking {
+      VStack {
+        Button("Increment") {
+          store.up.send(.increment)
+        }
+        Button("Decrement") {
+          store.down.send(.decrement)
+        }
+      }
+    }
+  }
+}
+```
+
+## Bindings
+
+For state properties that are writable, an automatic `Binding` can be derived by annotating the store with `@Bindable`. These bindings will use the workflow's state mutation sink. If you’re targeting iOS 16 or lower, you should use `@Perception.Bindable`.
+
+All properties can be turned into bindings by appending the `sending()` function to specify the “write” action. For properties that are already writable, this will refine the binding to send a custom action instead of the built-in state mutation sink.
+
+```swift
+@ObservableState
+struct State {
+   var isX: Bool
+   private(set) var isY: Bool
+}
+
+@CasePathable
+enum Action: WorkflowAction {
+  case setY(Bool)
+
+  func apply(toState state: inout State) -> Never? {
+    switch self {
+    case .setY(let value):
+      state.isY = value
+    }
+    return nil
+  }
+}
+
+typealias Model = ActionModel<State, Action>
+
+struct ContentView: View {
+  @Perception.Bindable var store: Store<Model>
+
+  var body: some View {
+    WithPerceptionTracking {
+      // synthesized binding
+      Toggle("X?", isOn: $store.isX)
+      // binding with a custom setter action
+      Toggle("Y?", isOn: $store.isY.sending(action: \.setY))
+    }
+  }
+}
+```
+
+## Parent dependencies
+
+Because `WorkflowSwiftUI` relies on observation to function, most properties on your model cannot be safely accessed directly through the `Store` — only the state, child models, and action sinks are accessible. So, when a workflow has dependencies provided by a parent (inputs), they must be added to your workflow’s own state in order to be visible by your view. They must also be updated in the `workflowDidChange` function of your workflow in order to receive updates from your parent.
+
+When updating dependencies, remember that any value being set is considered a mutation by the observation framework. To avoid invalidating your view on every render, check if dependencies have actually changed before copying them.
```

Comment:
Could we add some sort of operator to make this simpler, or a func to help with it? Eg, add a `~=` that checks the values before assigning them?